CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/layers/AbstractMapLayer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2010-2022 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.mapview2.layers;
+
+import au.gov.asd.tac.constellation.graph.Graph;
+import au.gov.asd.tac.constellation.views.mapview2.MapView;
+import javafx.scene.Group;
+
+/**
+ * Parent class to all layers
+ *
+ * @author altair1673
+ */
+public abstract class AbstractMapLayer {
+
+    protected MapView parent;
+    protected Graph currentGraph = null;
+    protected boolean isShowing = false;
+
+    protected int id;
+
+    protected AbstractMapLayer(final MapView parent, final int id) {
+        this.parent = parent;
+        currentGraph = parent.getCurrentGraph();
+        this.id = id;
+    }
+
+    // Class to hold a location in lattitude and longitude and its corresponding x and y values
+    protected class Location {
```

Comment:
doesn't seem like this class is being used within the AbstractMapLayer code itself. Might be more appropriate to put in its own class file?

Reply:
This class is used in the DayNightLayer and if in the future some Location data is needed for a map layer it kind of makes sense to keep it here. But moving it to the DayNightLayer class also makes sense?

Reply:
I think its more that the `Location` class seems like it's a completely separate class. `Location` holds no dependency on `AbstractMapLayer` and likewise, `AbstractMapLayer` holds no reference to `Location`. Therefore, it doesn't really make sense for `Location` to be an inner class of `AbstractMapLayer`

The implementation of `Location` itself is also doesn't seem specific to layers and so potentially could have other uses outside of that. That's why I suggest separating it out. The class itself doesn't really look like it should be an inner class